### DISCUSSION TITLE
Allure with Ctest

### DISCUSSION THREAD

**Discussion by pkp124** [allure-framework/allure2#2346] [Ideas & Feedback]
Hello,

Is it possible to implement an adaptor for CTest for allure. is there any documentation on how new adaptors can be implemented. Would you think this would be a good idea? Think there are many teams which CTest with C++ projects who could benefit from this.

Thanks

**Comment by baev** [comment]
CTest can generate test results in JUnit XML format, which is supported by Allure

**Reply by pkp124** [reply]
I meant, for generating history of tests and test results. Can that also be done with ctest?

**Reply by pkp124** [reply]
Hi @baev , any suggestions if it is possible to implement allure adapter for ctest. I am sorry if I am being annoying :)

**Reply by epszaw** [reply]
Yes, it could be. Just use JUnit XML format, then just generate the report for the results, the history feature will work as for another integrations.

**Reply by pkp124** [reply]
Thanks @baev , my understanding was that the test suite should generate a allure- results directory. Could you please show me( or point to the docs) how we can generate history of test results with only the report directory. I was not able to find it in the current documentation.
